services/github/github-api-provider.js

```diff
@@ -120,15 +178,12 @@ class GithubApiProvider {
       tokenString = this.globalToken
     }
 
-    const options = {
-      url,
-      baseUrl,
-      qs: query,
-      headers: {
-        'User-Agent': 'Shields.io',
-        Accept: 'application/vnd.github.v3+json',
-        Authorization: `token ${tokenString}`,
-      },
+    options.url = url
```

Comment:
Rather than mutate the input, could you copy it into a new object? It seems safer that way (in case e.g. an object is reused for that parameter).